stdlib/ballerina-builtin/src/main/ballerina/ballerina/auth/authz/authz-handler.bal

```diff
@@ -0,0 +1,102 @@
+
+@Description {value:"Authorization cache name"}
+const string AUTHZ_CACHE = "authz_cache";
+@Description {value:"Authentication header name"}
+const string AUTH_HEADER = "Authorization";
+@Description {value:"Basic authentication scheme"}
+const string AUTH_SCHEME = "Basic";
+
+@Description {value:"AuthorizationChecker instance"}
+AuthzChecker authzChecker;
+
+@Description {value:"Representation of Authorization Handler"}
+public struct HttpAuthzHandler {
+    // TODO
+}
+
+@Description {value:"Performs a authorization check, by comparing the groups of the user and the groups of the scope"}
+@Param {value:"req: InRequest instance"}
+@Param {value:"scopeName: name of the scope"}
+@Param {value:"resourceName: name of the resource which is being accessed"}
+@Return {value:"boolean: true if authorization check is a success, else false"}
+public function <HttpAuthzHandler httpAuthzHandler> handle (http:InRequest req,
+                                                                    string scopeName, string resourceName) (boolean) {
+
+    // TODO: extracting username and passwords are not required once the Ballerina SecurityContext is available
+    // extract the header value
+    var basicAuthHeaderValue, err = utils:extractBasicAuthHeaderValue(req);
+    if (err != null) {
+        log:printErrorCause("Error in extracting basic authentication header", err);
+        return false;
+    }
+
+    string username;
+    username, _, err = utils:extractBasicAuthCredentials(basicAuthHeaderValue);
+    if (err != null) {
+        log:printErrorCause("Error in decoding basic authentication header", err);
+        return false;
+    }
+    if (authzChecker == null) {
+        permissionstore:FileBasedPermissionStore fileBasedPermissionstore = {};
+        authzChecker = createChecker((permissionstore:PermissionStore)fileBasedPermissionstore,
+                                     utils:createCache(AUTHZ_CACHE));
+    }
+
+    // check in the cache. cache key is <username>-<resource>,
+    // since different resources can have different scopes
+    string authzCacheKey = username + "-" + resourceName;
+    any cachedAuthzResult = authzChecker.getCachedAuthzResult(authzCacheKey);
+    if (cachedAuthzResult != null) {
+        log:printDebug("Authz cache hit for user: " + username + ", request URL: " + resourceName);
+        var isAuthorized, typeCastErr = (boolean)cachedAuthzResult;
+        if (typeCastErr == null) {
+            // no type cast error, return cached result.
+            return isAuthorized;
+        }
+        // if a casting error occurs, clear the cache entry
+        authzChecker.clearCachedAuthzResult(authzCacheKey);
+    }
+    log:printDebug("Authz cache miss for user: " + username + ", request URL: " + resourceName);
+
+    boolean isAuthorized = authzChecker.check(username, scopeName);
+    if (isAuthorized) {
+        log:printInfo("Successfully authorized user: " + username + " for resource: " + resourceName);
```

Comment:
I don't think we need to log this as info since its a success scenario. Further more we should avoid logging the username at all times to be compliant with GDPR.